BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -358,6 +370,19 @@ public virtual async Task<HttpWebResponseWrapper> SendAsync()
             return Tuple.Create(transmissionA, transmissionB);
         }
 
+        /// <summary>
+        /// Serializes telemetry items.
+        /// </summary>
+        internal void Serialize(Uri address, IEnumerable<ITelemetry> telemetryItems, TimeSpan timeout = default(TimeSpan))
+        {
+            this.EndpointAddress = address;
```

Comment:
might be good to refactor (in a separate PR) this method, as this is doing more than serialization. (all the validations which were previously in ctor() is now done inside this method..)

Reply:
Added a TODO in summary.